### ISSUE TITLE
Different results between loacl device and huggingface-demo

### ISSUE THREAD

**Issue by BBO-Junbo** [swimmiing/ACL-SSL#4]
I tried some of my data on hugging face, the resullt was exciting! However, when I process the same audio-image pair at my own server, the result was different. The different results are shown as below.

![image](https://github.com/user-attachments/assets/86f43efd-0d81-426d-9fcf-371552e19313)
![1310](https://github.com/user-attachments/assets/8b63afe6-3f8a-4718-a823-2ab453b830b9)

**Comment by BBO-Junbo** [comment]
heat_map = process.draw_heatmap(Image.open(image_path).convert('RGB'), torchaudio.load(audio_path))

    def draw_heatmap(self, image, audio):
        # Get placeholder text
        prompt_template, text_pos_at_prompt, prompt_length = get_prompt_template()

        # Input pre processing
        audio, sample_rate = audio
        audio = audio.float() / 32768.0
        desired_sample_rate = 16000
        set_length = 10

        audio_file = audio

        if len(audio_file.shape) == 2:
            audio_file = torch.concat([audio_file[:, 0:1], audio_file[:, 1:2]], dim=0).T  # Stereo -> mono (x2 duration)
        else:
            audio_file = audio_file.unsqueeze(0)

        if desired_sample_rate != sample_rate:
            audio_file = torchaudio.functional.resample(audio_file, sample_rate, desired_sample_rate)

        audio_file = audio_file.squeeze(0)

        if audio_file.shape[0] > (desired_sample_rate * set_length):
            audio_file = audio_file[:desired_sample_rate * set_length]

        # zero padding
        if audio_file.shape[0] < (desired_sample_rate * set_length):
            pad_len = (desired_sample_rate * set_length) - audio_file.shape[0]
            pad_val = torch.zeros(pad_len)
            audio_file = torch.cat((audio_file, pad_val), dim=0)

        audio_file = audio_file.unsqueeze(0)

        image_transform = vt.Compose([
            vt.Resize((352, 352), vt.InterpolationMode.BICUBIC),
            vt.ToTensor(),
            vt.Normalize((0.48145466, 0.4578275, 0.40821073), (0.26862954, 0.26130258, 0.27577711)),  # CLIP
        ])

        image_file = image_transform(image).unsqueeze(0)

        # Inference
        placeholder_tokens = self.model.get_placeholder_token(prompt_template.replace('{}', ''))
        audio_driven_embedding = self.model.encode_audio(audio_file.to(self.model.device), placeholder_tokens, text_pos_at_prompt,
                                                    prompt_length)

        # Localization result
        out_dict = self.model(image_file.to(self.model.device), audio_driven_embedding, 352)
        seg = out_dict['heatmap'][0:1]
        seg_image = ((1 - seg.squeeze().detach().cpu().numpy()) * 255).astype(np.uint8)
        seg_image = Image.fromarray(seg_image)

        seg_image = seg_image.resize(image.size, Image.BICUBIC)
        heatmap_image = cv2.applyColorMap(np.array(seg_image), cv2.COLORMAP_JET)
        overlaid_image = cv2.addWeighted(np.array(image), 0.5, heatmap_image, 0.5, 0)

        return overlaid_image

**Comment by swimmiing** [comment]
This issue has been resolved via email. The problem was caused by incorrect audio normalization in the code provided above.
